float32 lat
float32 lng
float32 angle
string state
bool loop
string pathID
int targetIndex